assets/global.js

```diff
@@ -443,6 +443,7 @@ class ModalDialog extends HTMLElement {
       'click',
       this.hide.bind(this, false)
     );
+    ['shopify:block:select', 'shopify:section:select'].forEach(event => document.addEventListener(event, () => this.hide()));
```

Comment:
Do we need to add more events here, like `shopify:section:deselect` and `shopify:block:deselect`?